### PR TITLE
[KOJI-123] Data Grid

### PR DESCRIPTION
**Pull request by eynglv**
`DataGrid` component that leverages [Mui's DataGrid](https://mui.com/x/api/data-grid/data-grid/), which we'll use for [Leads Management Page](https://gethearth.atlassian.net/jira/software/projects/KOJI/boards/36?selectedIssue=KOJI-123#:~:text=leads%20management%20page)
 
Screen Shot 2022-11-14 at 2 31 18 PM

**Jira Ticket:** https://gethearth.atlassian.net/browse/KOJI-123

### CODE REVIEW

**Review by brianclee-gh — APPROVED** [review]

File: src/components/DataGrid/DataGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+import theme from '../../theme';
+import { makeStyles } from '../../theme';
+import Box from '../Box';
+
+export interface Props extends DataGridProps {
+  columns: GridColDef[];
+  rows: GridRowsProp;
+  pageSize?: number;
+  height: string;
```
Comment (smurmus) [inline src/components/DataGrid/DataGrid.tsx:17]: the former two are more straightforward but could you add comments above these two props for what they do? (`pageSize` and `height`)

Comment (smurmus) [reply src/components/DataGrid/DataGrid.tsx:17]: and also using the `@default` syntax we use in a few places for the `pageSize` prop

File: src/components/DataGrid/DataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+const useStyles = makeStyles({
```
Comment (smurmus) [inline src/components/DataGrid/DataGrid.tsx:20]: should be able to use `styled` here; don't want to use `makeStyles` anymore!

the css classes should be able to be used the same way

**Review by smurmus — CHANGES_REQUESTED** [review]

File: src/components/DataGrid/DataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { DataGrid as MuiDataGrid, DataGridProps } from '@mui/x-data-grid';
+
+import theme from '../../theme';
+import { makeStyles } from '../../theme';
+import Box from '../Box';
+import styled from '../../theme/styled';
+
+export interface Props extends DataGridProps {
```
Comment (smurmus) [inline src/components/DataGrid/DataGrid.tsx:9]: should do `export interface DataGridProps extends MuiDataGridProps` (import native props with the Mui prefix)

**Review by smurmus — APPROVED** [review]
small comment around props naming!

**Pull request merged**